market_maker_keeper/uniswapv2_market_maker_keeper.py

```diff
@@ -0,0 +1,441 @@
+from market_maker_keeper.feed import ExpiringFeed, WebSocketFeed
+from pymaker.keys import register_keys
+from pymaker import Address, Wad, Receipt
+from market_maker_keeper.gas import GasPriceFactory
+from market_maker_keeper.model import TokenConfig
+from market_maker_keeper.reloadable_config import ReloadableConfig
+
+class UniswapV2MarketMakerKeeper:
+    """Keeper acting as a market maker on UniswapV2.
+
+    Adding or removing liquidity to arbitrary ETH-ERC20 or ERC20-ERC20 Pools.
+    If one of the assets is ETH, it is assumed that it will be the second asset (token_b)
+
+    """
+
+    logger = logging.getLogger()
+
+    def __init__(self, args: list, **kwargs):
+        parser = argparse.ArgumentParser(prog='uniswap-market-maker-keeper')
+
+        parser.add_argument("--rpc-host", type=str, default="localhost",
+                            help="JSON-RPC host (default: `localhost')")
+
+        parser.add_argument("--rpc-port", type=int, default=8545,
+                            help="JSON-RPC port (default: `8545')")
+
+        parser.add_argument("--rpc-timeout", type=int, default=10,
+                            help="JSON-RPC timeout (in seconds, default: 10)")
+
+        parser.add_argument("--eth-from", type=str, required=True,
+                            help="Ethereum account from which to send transactions")
+
+        parser.add_argument("--eth-key", type=str, nargs='*',
+                            help="Ethereum private key(s) to use (e.g. 'key_file=aaa.json,pass_file=aaa.pass')")
+
+        parser.add_argument("--pair", type=str, required=True,
+                            help="Token pair (sell/buy) on which the keeper will operate")
+
+        parser.add_argument("--token-config", type=str, required=True,
+                            help="Token configuration file")
+
+        parser.add_argument("--uniswap-feed", type=str, required=True,
+                            help="Source of liquidity feed")
+
+        parser.add_argument("--uniswap-feed-expiry", type=int, default=86400,
+                            help="Maximum age of the liquidity feed (in seconds, default: 86400)")
+
+        parser.add_argument("--gas-price", type=int, default=9000000000,
+                            help="Gas price (in Wei)")
+
+        parser.add_argument("--smart-gas-price", dest='smart_gas_price', action='store_true',
+                            help="Use smart gas pricing strategy, based on the ethgasstation.info feed")
+
+        parser.add_argument("--initial-exchange-rate", type=float, default=None,
+                            help="Used to determine the initial ratio to be used in a newly created pool")
+
+        parser.add_argument("--percentage-difference", type=float, default=2,
+                            help="Percentage difference between Uniswap exchange rate and aggregated price"
+                                 "(default: 2)")
+
+        parser.add_argument("--uniswap-percentage-difference", type=float, default=5,
```

Comment:
not quite sure what this one does... if I understand correctly we should just need accepted slippage.... could you unpack this one for me @MikeHathaway ?

Reply:
This is older code that was used to predict future price movements based on the mempool. I've removed it as its now unused.